packages/sync-service/lib/electric/schema.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Electric.Schema do
+  import Bitwise
+
+  @bit_types ["bit", "varbit"]
+  @variable_length_character_types ["varchar", "text"]
+  @fixed_length_character_types ["bpchar"]
+  @time_types ["time", "timetz", "timestamp", "timestamptz", "interval", "date"]
+  @interval_field_masks [
+    %{unit: "YEAR", mask: 1 <<< 2},
+    %{unit: "MONTH", mask: 1 <<< 1},
+    %{unit: "DAY", mask: 1 <<< 3},
+    %{unit: "HOUR", mask: 1 <<< 10},
+    %{unit: "MINUTE", mask: 1 <<< 11},
+    %{unit: "SECOND", mask: 1 <<< 12}
+  ]
+  # -1 encoded as a 16 bit integer
+  @minus_1 65535
+  # all bits set in a 16 bit integer
+  @all_set 32767
+
+  def from_column_info(column_info) do
+    Map.new(column_info, fn col -> {col.name, schema(col)} end)
+  end
+
+  defp schema(%{array_dimensions: array_dimensions} = col_info) do
+    %{
+      type: type(col_info),
+      dimensions: array_dimensions
```

Comment:
I'd rather use a shorter key here given we're shoving it into every column repesentation, say `dims`, but maybe just make that key not persent for non-array types?

Reply:
Will shorten it to `dims`. Regarding leaving it out for non-array types, i proposed the same yesterday but @robacourt  prefers always having it to avoid confusion. His argument being that by having it always present it forces the oen that writes a parser to think about array types, which otherwise they may not do from the get go.

Reply:
I'd argue that making headers on every single request slightly smaller has a larger benefit than one-time library author interaction, but I'm not going to push it further

Reply:
I've changed my mind about always having the dimensions attribute in the type schema. My initial concerns although probably valid don't outweigh the fact that it's a surprising, inelegant and inefficient attribute to have for a non array type.